torch/_inductor/sizevars.py

```diff
@@ -307,6 +311,19 @@ def statically_known_multiple_of(self, numerator: Expr, denominator: Expr) -> bo
         expr = sympy.Eq(numerator % denominator, 0)
         return self.is_expr_static_and_true(expr)
 
+    # The guard functions require you to ALREADY KNOW that a particular
+    # condition holds.  If you don't know (you want to guard on an expression
+    # being a particular value, and then get access to that value), use
+    # the evaluate functions.
+
```

Comment:
nit newline? also maybe we should just always go through speculation? requiring that they already know can be an assert at the callsite...?